src/services/BookingService.ts

```diff
@@ -119,13 +121,48 @@ class BookingService {
     return booking;
   }
 
-  public async deleteById(id: number): Promise<void> {
+  public async bookingBelongsToTheUser(
+    bookingId: number, userId: number,
+  ): Promise<boolean> {
+    const repo = this.getRepository();
+    const booking = await repo.findOneBy({ id: bookingId });
+
+    console.log("Booking:");
+    console.log(booking);
+
+    if (Number(booking.user) === userId) {
+      return true;
+    }
+
+    return false;
+  }
+
+  public async deleteById(
+    bookingId: number, userId: number, userRole: string,
+  ): Promise<void> {
+    const isBookingBelongToTheUser = await this.bookingBelongsToTheUser(
```

Comment:
It's not bad to have this method defined in the service, but we should not make the decision whether it has authorization or not here, we have the authorization middleware for that.